addons/DistancePlus/DistancePlus.lua

```diff
@@ -0,0 +1,420 @@
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'DistancePlus'
+_addon.author = 'Sammeh'
+_addon.version = '1.3.0.7'
+_addon.command = 'dp'
+
+-- 1.3.0.2 Fixed up nil's per recommendation on submission to Windower 
+-- 1.3.0.3 Replaced all tabs for 4 spaces to normalize indentations.
+-- 1.3.0.4 Moving some expensive functions to on-load vs per-render.
+-- 1.3.0.5 Implement config plugin.
+-- 1.3.0.6 Fix ability list on job change.
+-- 1.3.0.7 Implement ranged fix w/o ja_distance
+
+require('tables')
+
+res = require 'resources'
+config = require('config')
+texts = require('texts')
+
+defaults = {}
+defaults.main = {}
+defaults.main.pos = {}
+defaults.main.pos.x = -178
+defaults.main.pos.y = 21
+defaults.main.text = {}
+defaults.main.text.font = 'Arial'
+defaults.main.text.size = 14
+defaults.main.flags = {}
+defaults.main.flags.right = true
+
+defaults.pettxt = {}
+defaults.pettxt.pos = {}
+defaults.pettxt.pos.x = -178
+defaults.pettxt.pos.y = 45
+defaults.pettxt.text = {}
+defaults.pettxt.text.font = 'Arial'
+defaults.pettxt.text.size = 14
+defaults.pettxt.flags = {}
+defaults.pettxt.flags.right = true
+
+
+defaults.abilitytxt = {}
+defaults.abilitytxt.pos = {}
+defaults.abilitytxt.pos.x = -80
+defaults.abilitytxt.pos.y = 45
+defaults.abilitytxt.text = {}
+defaults.abilitytxt.text.font = 'Arial'
+defaults.abilitytxt.text.size = 10
+defaults.abilitytxt.flags = {}
+defaults.abilitytxt.flags.right = true
+
+defaults.heighttxt = {}
+defaults.heighttxt.pos = {}
+defaults.heighttxt.pos.x = -238
+defaults.heighttxt.pos.y = 21
+defaults.heighttxt.text = {}
+defaults.heighttxt.text.font = 'Arial'
+defaults.heighttxt.text.size = 14
+defaults.heighttxt.flags = {}
+defaults.heighttxt.flags.right = true
+
+height_upper_threshold = 8.5
+height_lower_threshold = -7.5
+
+
+settings = config.load(defaults)
+distance = texts.new('${value||%.2f}', settings.main)
+petdistance = texts.new('${value||%.2f}', settings.pettxt)
+abilities = texts.new('${value}', settings.abilitytxt)
+height = texts.new('${value||%.2f}', settings.heighttxt)
+
+
+option = "Default"
+showabilities = false
+showheight = false
+
+function displayabilities(distance,master_pet_distance)
+    local range_mult = {
+        [2] = 1.55,
+        [3] = 1.490909,
+        [4] = 1.44,
+        [5] = 1.377778,
+        [6] = 1.30,
+        [7] = 1.15,
+        [8] = 1.25,
+        [9] = 1.377778,
+        [10] = 1.45,
+        [11] = 1.454545454545455,
+        [12] = 1.666666666666667,
+    }
+
+    list = 'Abilities:\n'
+    if abilitylist then 
+      for key,ability in pairs(abilitylist) do
+        ability_en = res.job_abilities[ability].en
```

Comment:
Two things when using resources, first be aware that `en` is not meant to be used - use `english` instead.

Second, if you use `name` instead of `english` (or `en`) the name will display correctly on Japanese clients as well, so you'd be doing them a favor.